packages/website/src/snippets.ts

```diff
@@ -31,96 +31,78 @@ function defaultTo(
-            <BaseText>Through \${data.endDate}</BaseText>
-            
-            <BaseText faded>Location</BaseText>
-            <BaseText>\${data.galleryLocation}</BaseText>
-            
-            <BaseText faded>About this exhibition</BaseText>
-            <BaseText>\${data.summary}</BaseText>
-        \`;
-    };
-  
-  
-    const touchpoint = await create({
-        config: {
-          applicationUrl: "${config.applicationUrl ?? "REPLACE_WITH_APPLICATION_URL"}",
-          headers: {
-            "nlx-api-key": "${config.headers?.["nlx-api-key"] ?? "REPLACE_WITH_API_KEY"}"
-          },
-          languageCode: "${config.languageCode ?? "en-US"}",
-          userId: "${config.userId ?? "REPLACE_WITH_USER_ID"}"
-        },
-        colorMode: "dark",
-        input: "text",
-        theme: { "fontFamily": "\\"Neue Haas Grotesk\\", sans-serif", "accent": "#AECAFF" },
-        customModalities: {
-            MuseumExhibitDetails: MuseumExhibitDetails,
-            MuseumExhibitCarousel: MuseumExhibitCarousel
-        }
-    });`;
+const museumComponents = `
+const MuseumExhibitCarousel = ({ data, conversationHandler }) => {
+  const [selected, setSelected] = React.useState(null);
+
+  return html\`
+    <Carousel>
+      \${data.map((exhibit, index) => html\`
+        <CustomCard
+          key=\${index}
+          selected=\${selected === index}
+          onClick=\${() => {
+          setSelected(index);
+          conversationHandler.sendChoice(exhibit.id);
+          }}
+        >
+          <CustomCardImageRow 
+            src=\${exhibit.imageUrl} 
+            alt=\${exhibit.name}
+          />
+          <CustomCardRow
+            left=\${html\`<BaseText faded><div/></BaseText>\`}
+            right=\${html\`<BaseText>\${exhibit.name}</BaseText>\`}
+          />
+          <CustomCardRow
+            left=\${html\`<BaseText faded>Dates:</BaseText>\`}
+            right=\${html\`<BaseText>Through \${exhibit.endDate}</BaseText>\`}
+          />
+        </CustomCard>\`
+      )}
+    </Carousel>
+  \`;
+};
+
+const MuseumExhibitDetails = ({ data, conversationHandler }) => {
+  const detailedUrls = data.detailImageUrls;
+  return html\`
+    <Carousel>
+      <CustomCard>
+        <CustomCardImageRow 
+          src=\${data.imageUrl} 
+          alt=\${data.name}
+        />
+      </CustomCard>
+      \${detailedUrls.map((imageUrl) => html\`
+        <CustomCard>
+          <CustomCardImageRow 
+            src=\${imageUrl} 
+            alt=\${data.name}
+          />
+        </CustomCard>\`
+      )}
+    </Carousel>
+    <BaseText faded>Dates</BaseText>
+    <BaseText>Through \${data.endDate}</BaseText>
+        
+    <BaseText faded>Location</BaseText>
+    <BaseText>\${data.galleryLocation}</BaseText>
+        
+    <BaseText faded>About this exhibition</BaseText>
+    <BaseText>\${data.summary}</BaseText>
+  \`;
 };
+`;
+
+export type TemplateComponents = "noComponents" | "museumComponents";
```

Comment:
Same as mentioned before, since this paradigm will likely grow, we should expect multiple templates.